x/ethbridge/module.go

```diff
@@ -0,0 +1,145 @@
+package ethbridge
+
+import (
+	"encoding/json"
+
+	"github.com/cosmos/peggy/x/ethbridge/client"
+	"github.com/cosmos/peggy/x/ethbridge/types"
+	"github.com/cosmos/peggy/x/oracle"
+	"github.com/gorilla/mux"
+	"github.com/spf13/cobra"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+)
+
+var (
+	_ module.AppModule      = AppModule{}
+	_ module.AppModuleBasic = AppModuleBasic{}
+)
+
+// AppModuleBasic defines the basic application module used by the ethbridge module.
+type AppModuleBasic struct{}
+
+var _ module.AppModuleBasic = AppModuleBasic{}
+
+// Name returns the ethbridge module's name.
+func (AppModuleBasic) Name() string {
+	return ModuleName
+}
+
+// RegisterCodec registers the ethbridge module's types for the given codec.
+func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {
+	types.RegisterCodec(cdc)
+}
+
+// DefaultGenesis returns default genesis state as raw bytes for the ethbridge
+// module.
+func (AppModuleBasic) DefaultGenesis() json.RawMessage {
+	return nil
+}
+
+// ValidateGenesis performs genesis state validation for the ethbridge module.
+func (AppModuleBasic) ValidateGenesis(_ json.RawMessage) error {
+	return nil
+}
+
+// RegisterRESTRoutes registers the REST routes for the ethbridge module.
+func (AppModuleBasic) RegisterRESTRoutes(ctx context.CLIContext, rtr *mux.Router) {
+	client.RegisterRESTRoutes(ctx, rtr, StoreKey)
+}
+
+// GetTxCmd returns the root tx command for the ethbridge module.
+func (AppModuleBasic) GetTxCmd(cdc *codec.Codec) *cobra.Command {
+	return client.GetTxCmd(StoreKey, cdc)
+}
+
+// GetQueryCmd returns no root query command for the ethbridge module.
+func (AppModuleBasic) GetQueryCmd(cdc *codec.Codec) *cobra.Command {
+	return client.GetQueryCmd(StoreKey, cdc)
+}
+
+//____________________________________________________________________________
+
+// AppModuleSimulation defines the module simulation functions used by the ethbridge module.
+type AppModuleSimulation struct{}
```

Comment:
I think you can delete the `AppModuleSimulation` for now

Reply:
will leave this in for now as we'll start working on it soon anyway